src/Microsoft.DotNet.Wpf/specs/tooltip.md

```diff
@@ -0,0 +1,305 @@
+Editor activity includes mouse clicks or keystrokes on the editing surface, and `UpdateComposition` requests from an IME.
+- **Exclusion**.  Opening a tooltip or context menu (for any reason) closes the previously open tooltip, if any.
+- **Programmatically**.  [TODO: check if this is possible. If so, describe it.]
+
+## Mouse buffer zone
+When you open a tooltip by hovering the mouse over its parent element, you should be able to move the mouse onto the tooltip without closing the tooltip.
+To that end, WPF implicitly declares a *mouse buffer zone*: a rectangle that connects the parent element to the tooltip, in which you can move the mouse without closing the tooltip due to hover-end.
+(Note that the mouse still interacts in the usual way with elements in the buffer zone, and that interaction might close the tooltip.)
+
+After using the usual placement rules to determine the size and position of the tooltip, WPF calculates the buffer zone to be a rectangle with the following properties:
+- it contains the shortest straight line between the parent and its tooltip
+- it is large enough to accommodate reasonable drift as you move the mouse along that line
+- it is no larger than needed to satisfy the first two properties
+
+In the common case where the tooltip overlaps its parent, the buffer zone may be empty.
+
+It may be impossible or impractical to find a suitable rectangle, in which case the buffer zone is empty and moving the mouse off the parent element will close the tooltip.
+This only happens in extreme cases, for example when the tooltip and its parent are on different monitors. 
+
+
+# API Pages
+
+## ToolTipService.ShowsToolTipOnKeyboardFocus attached property
+Gets or sets whether an element shows its tooltip when it acquires keyboard focus.
+
+```
+see GetShowsToolTipOnKeyboardFocus, and SetShowsToolTipOnKeyboardFocus
+```
+
+### Remarks
+Setting this property on a parent element that has a tooltip helps control whether to open the tooltip when the element acquires focus by keyboard navigation.
+(Note: when the parent acquires focus by mouse-click or touch, the tooltip is not opened.)
+The tooltip is opened according to the following table, where the rows indicate the value of the `ToolTipService.ShowsToolTipOnKeyboardFocus` attached property on the parent element, and the columns indicate the value of the ToolTip element's `ShowsToolTipOnKeyboardFocus` property:
+
+|           | False | null  | True  |
+|-----------| ----- | ----  | ----  |
+| **False** | False | False | False |
+| **null**  | False | True  | True  |
+| **True**  | True  | True  | True  |
+
+If there is no ToolTip element, the 'null' column applies.
+
+In other words, honor an explicit value (True or False); if both the parent element and the ToolTip element have explicit values, the parent's value takes precedence;  if neither has an explicit value, open the tooltip.  The latter case is the default.
```

Comment:
This is great and succinct, I would leave out the table.

Reply:
In the real doc pages, perhaps.  Although I bet there are people who prefer the table to the text.  

In the spec, the table is more precise and definitive, less likely to be misinterpreted by the implementer, tester, or compliance reviewer.